src/libraries/System.Text.Json/gen/src/JsonSourceGeneratorHelper.Generate.cs

```diff
@@ -0,0 +1,677 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text.Json.Serialization;
+using System.Text.Json.SourceGeneration.Reflection;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+
+namespace System.Text.Json.SourceGeneration
+{
+    internal sealed partial class JsonSourceGeneratorHelper
+    {
+        private readonly string _generationNamespace;
+
+        // TODO: consider public option for this.
+        private bool _honorRuntimeProvidedCustomConverters = true;
+
+        private const string RuntimeCustomConverterFetchingMethodName = "GetRuntimeProvidedCustomConverter";
+
+        private const string JsonContextDeclarationSource = "internal partial class JsonContext : JsonSerializerContext";
+
+        private const string OptionsInstanceVariableName = "Options";
+
+        private const string PropInitFuncVarName = "PropInitFunc";
+
+        private const string JsonMetadataServicesClassName = "JsonMetadataServices";
+
+        private const string CreateValueInfoMethodName = "CreateValueInfo";
+
+        // TODO: Enable localization for these descriptors.
```

Comment:
Is there an issue tracking this?

Reply:
As a nit, we generally use Issues to track to-dos, not our own code. The reasons include clutter, duplication, and they tend to not get cleaned up. An exception might be when noting potential future improvement makes it easier to understand the code.

Reply:
https://github.com/dotnet/runtime/issues/45448#issuecomment-822144624